src/preferences/common/preference-service.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { Event, Emitter } from '../../application/common/event'
+import { PreferenceChangedEvent } from './preference-event'
+import { IPreferenceServer } from './preference-server'
+
+export const IPreferenceService = Symbol("IPreferenceService")
+
+
+injectable()
+export interface IPreferenceService {
+    readonly onPreferenceChanged: Event<PreferenceChangedEvent>;
+
+    has(preferenceName: string): Promise<boolean>;
+
+    get<T>(preferenceName: string): Promise<T | undefined>;
+    get<T>(preferenceName: string, defaultValue?: T): Promise<T>;
+
+
+    getBoolean(preferenceName: string): Promise<boolean | undefined>;
+    getBoolean(preferenceName: string, defaultValue?: boolean): Promise<boolean>;
+
+
+    getString(preferenceName: string): Promise<string | undefined>;
+    getString(preferenceName: string, defaultValue?: string): Promise<string>;
+
+
+    getNumber(preferenceName: string): Promise<number | undefined>;
+    getNumber(preferenceName: string, defaultValue?: number): Promise<number>;
+
+}
+
+export interface IPreferenceClient {
```

Comment:
it should not be here, but go next to `IPreferenceServer`, e.g. in `preference-protocol.ts`